sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -934,79 +705,27 @@ def download_blob(
                 :dedent: 12
                 :caption: Download a blob.
         """
-        options = self._download_blob_options(
+        if self.require_encryption and not (self.key_encryption_key or self.key_resolver_function):  # pylint: disable=line-too-long
+            raise ValueError("Encryption required but no key was provided.")
+        if length is not None and offset is None:
+            raise ValueError("Offset value must not be None if length is set.")
+        if kwargs.get('cpk') and self.scheme.lower() != 'https':
+            raise ValueError("Customer provided encryption key must be used over HTTPS.")
+        options = _download_blob_options(
+            encryption_options=self._build_encryption_options(),
+            version_id=get_version_id(self.version_id, kwargs),
+            sdk_moniker=self._sdk_moniker,
+            encryption_version=self.encryption_version,
+            client=self._client,
+            config=self._config,
+            blob_name=self.blob_name,
+            container_name=self.container_name,
             offset=offset,
             length=length,
             encoding=encoding,
             **kwargs)
```

Comment:
Same comment about encryption options here, also you `encryption_version` twice now

Also suggested ordering again :)
```py
blob_name=self.blob_name,
container_name=self.container_name,
version_id=get_version_id(self.version_id, kwargs),
offset=offset,
length=length,
encoding=encoding,
encryption_options={
    'required': self.require_encryption,
    'version': self.encryption_version,
    'key': self.key_encryption_key,
    'resolver': self.key_resolver_function
},
config=self._config,
sdk_moniker=self._sdk_moniker,
client=self._client,
**kwargs
```